Source/Dafny/Rewriter.cs

```diff
@@ -118,6 +119,36 @@ public abstract class IRewriter {
       Contract.Requires(program != null);
     }
   }
+  // intern project 
+
+  class tobiRewriter : IRewriter {
+    internal override void PostResolve(Program program) {
+      base.PostResolve(program);
+      foreach (var moduleDefinition in program.Modules()) {
+        foreach (var topLevelDecl in moduleDefinition.TopLevelDecls.OfType<TopLevelDeclWithMembers>()) {
+          foreach (var function in topLevelDecl.Members.OfType<Function>()) {
+            if (function.Body is NestedMatchExpr matchExpr) {
```

Comment:
This only matches on match expressions at the root level of a function, not in expressions that occur in arbitrary places.